modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/WorkflowInstance.java

```diff
@@ -82,6 +82,32 @@
                 name = "Workflow.findAll",
                 query = "select w from WorkflowInstance w where w.organizationId=:organizationId order by w.dateCreated"
         ),
+        // Finding the latest or active workflow for all events
+        @NamedQuery(
+                name = "Workflow.findLatest",
+                query = "SELECT w FROM WorkflowInstance w "
```

Comment:
Looking at this, I'm wondering if we shouldn't go with native data types right away and just ditch the in database and just do something like 
```sql
SELECT id, mediapackage_id, state FROM oc_workflow order by date_created;
```
Getting these data would be simpler, faster and would probably even take less memory (compare with this approach):

```
opencast=# EXPLAIN ANALYSE
SELECT w.id, w.mediapackage_id, w.state FROM oc_workflow w
where w.date_created = (SELECT MAX(date_created) FROM oc_workflow w2 WHERE w2.mediapackage_id = w.mediapackage_id);
                                                                           QUERY PLAN                                                                           
----------------------------------------------------------------------------------------------------------------------------------------------------------------
 Seq Scan on oc_workflow w  (cost=0.00..64112.57 rows=38 width=49) (actual time=0.051..63.822 rows=7513 loops=1)
   Filter: (date_created = (SubPlan 1))
   Rows Removed by Filter: 44
   SubPlan 1
     ->  Aggregate  (cost=8.30..8.31 rows=1 width=8) (actual time=0.007..0.007 rows=1 loops=7557)
           ->  Index Scan using ix_oc_workflow_mediapackage_id on oc_workflow w2  (cost=0.28..8.30 rows=1 width=8) (actual time=0.005..0.006 rows=1 loops=7557)
                 Index Cond: ((mediapackage_id)::text = (w.mediapackage_id)::text)
 Planning Time: 0.230 ms
 Execution Time: 64.442 ms
(9 rows)

opencast=# EXPLAIN ANALYSE
SELECT id, mediapackage_id, state FROM oc_workflow order by date_created;
                                                     QUERY PLAN                                                      
---------------------------------------------------------------------------------------------------------------------
 Sort  (cost=1647.52..1666.45 rows=7571 width=57) (actual time=7.998..8.725 rows=7557 loops=1)
   Sort Key: date_created
   Sort Method: quicksort  Memory: 1255kB
   ->  Seq Scan on oc_workflow  (cost=0.00..1159.71 rows=7571 width=57) (actual time=0.011..4.259 rows=7557 loops=1)
 Planning Time: 0.084 ms
 Execution Time: 9.925 ms
(6 rows)
```

Reply:
As per discussion: For another pull request…